### ISSUE TITLE
Use symbols correctly: code fix bug with null-conditional access operator

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#4337]
Caught this while applying the code fix to:

https://github.com/dotnet/roslyn-analyzers/blob/91b109f2f69496b76e7a740a4547a9d9b9a12c0b/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L296

It was transformed into:

```csharp
if (self? SymbolEqualityComparer.Default.Equals(self, overload) == true)
```

**Comment by Youssef1313** [comment]
The analyzer itself is reporting on wrong span `a?.[|Equals(b)|]`
It should report on the whole expression `[|a?.Equals(b)|]`

Not sure if the codefix will behave correctly after updating the analyzer or not.
I'll look into it.

**Comment by Youssef1313** [comment]
I found it easier to keep the diagnostic span as-is and update the fixer. Otherwise, this will require updating both the analyzer and the fixer.
